### PR TITLE
feat(view): added iOS parameter for modal presentation style

### PR DESCRIPTION
**Pull request by surdu**
## PR Checklist

- [x] The PR title follows our guidelines: https://github.com/NativeScript/NativeScript/blob/master/CONTRIBUTING.md#commit-messages.
- [x] There is an issue for the bug/feature this PR is for. To avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it.
- [x] You have signed the [CLA](http://www.nativescript.org/cla).
- [ ] All existing tests are passing: https://github.com/NativeScript/NativeScript/blob/master/DevelopmentWorkflow.md#running-unit-tests.
- [ ] Tests for the changes are included - https://github.com/NativeScript/NativeScript/blob/master/WritingUnitTests.md.

## What is the current behavior?
Other than fullscreen or not, there is no way to set the rest of modal presentation styles for a modal on iOS

## What is the new behavior?
You can now pass an extra parameter called `iOSPresentationStyle` to `.showModal()`

Implements #6387

### CODE REVIEW

**Comment by surdu** [comment]
@tsonevn It turns out this is not so easy as just passing a parameter to `.showModal()`. Some of the presentation styles require additional logic in order to make them work.

I've already implemented [some default logic](https://github.com/NativeScript/NativeScript/pull/6409/files#diff-2775a71872c1630de8108bf058a1628eR403) for `UIModalPresentationPopover` and while I believe some default logic can also be implemented for most of the otehrs, `UIModalPresentationCustom` might be a little bit complicated withouth returning the view controller to the user.

I will investigate further, but wanted to open the discussion here for further ideas.

**Comment by vakrilov** [comment]
Hey there @surdu - thanks for you effort. Here are a couple of thoughts:

### Definitions
First of all - all `d.ts` files should be clear of IOS or Android Specific classes(like `UIModalPresentationStyle`). The reason is that we don't want people to be forced to use `tna-platform-declarations` package in their apps.

### Options
The `showModal` API already has 3 optional parameters and this is making it little hard to use. Adding one more optional param will amplify the problem (what if you want to pass only the last one). 

My suggestion is to define a `ShowModalOptions` interface that you pass to the method. This way you can fill in only the options you care about. Furthermore you can have an `ios` and `android` portion of the interface that only deals with plat-spceific options (like `UIModalPresentationStyle`). 

You can also have a callback in the options - which will be called if they are using the an option, that needs to give the control back to the user for additional setup of a view-controller for example.

### PR
Seems that this PR is not as easy as originally thought. What do you think of moving the discussion back to the issue (#6387). We might pick only the `UIModalPresentationPopover` values that are most useful - no point of supporting all of them if they are not going to be used.

**Comment by surdu** [comment]
@vakrilov I've made the changes that you requested.

Also wrote a test for my changes. I took another test for the modal and just changed the way the options are passed. The test is somewhat overkill, but I guess there was a good reason to implement the test I copied like this, so I went with it.

There is one thing that I want to do but I didn't manage to figure out (probably can't be done at this stage of execution): I want to get styling informations about the Page that is about to be presented [at this point](https://github.com/NativeScript/NativeScript/pull/6409/files#diff-2775a71872c1630de8108bf058a1628eR405). It would be nice to get the Page width and height expressed by the user is css and resize the pop-up accordingly. Also needed the background color few lines bellow.

For the old version of `showModal` that receives the options as arguments, do you want to deprecate it or will they live alongside each other?

**Comment by vakrilov** [comment]
test

**Comment by dtopuzov** [comment]
test

File: tns-core-modules/ui/core/view/view-common.ts [hunk 1]
```diff
@@ -243,10 +243,10 @@ export abstract class ViewCommon extends ViewBase implements ViewDefinition {
           stretched = arguments[5];
         }
 
-        const view: ViewDefinition = firstAgrument instanceof ViewCommon
-            ? firstAgrument : createViewFromEntry({ moduleName: firstAgrument });
+        const view = firstAgrument instanceof ViewCommon
+            ? firstAgrument : <ViewCommon>createViewFromEntry({ moduleName: firstAgrument });
 
-        (<ViewCommon>view)._showNativeModalView(this, context, closeCallback, fullscreen, animated, stretched, iosOpts);
```
Comment (surdu) [inline tns-core-modules/ui/core/view/view-common.ts]: Note to self: removed `iosOpts` by mistake

**Comment by vakrilov** [comment]
test

**Comment by vakrilov** [comment]
test

**Comment by vakrilov** [comment]
test

**Pull request merged**